templates/triton_services/Dockerfile

```diff
@@ -0,0 +1,61 @@
+FROM nvcr.io/nvidia/tritonserver:24.04-py3
+
+#### Start of Stable Diffusion modifications ####
+# Install dependencies related to build and serve stable diffusion
+RUN pip install --no-cache-dir tritonclient[all]==2.45.0 torch==2.3.0 diffusers==0.23.1 \
+    onnx==1.14.0 onnx-graphsurgeon==0.5.2 polygraphy==0.49.9 transformers==4.31.0 scipy==1.13.0 \
+    nvtx==0.2.10 accelerate==0.30.1 optimum[onnxruntime]==1.19.2 nvidia-ammo==0.9.4
+
+# Install Triton python api, copy model.py, and model config
+RUN git clone https://github.com/triton-inference-server/tutorials.git -b r24.04 --single-branch /tmp/tutorials
+RUN pip --no-cache-dir install /tmp/tutorials/Triton_Inference_Server_Python_API/deps/tritonserver-2.41.0.dev0-py3-none-any.whl[all]
+RUN mkdir -p /opt/tritonserver/backends/diffusion
+RUN cp /tmp/tutorials/Popular_Models_Guide/StableDiffusion/backend/diffusion/model.py /opt/tritonserver/backends/diffusion/model.py
+RUN mkdir -p /tmp/workspace/diffusion-models/stable_diffusion_1_5/1
+RUN cp /tmp/tutorials/Popular_Models_Guide/StableDiffusion/diffusion-models/stable_diffusion_1_5/config.pbtxt /tmp/workspace/diffusion-models/stable_diffusion_1_5/config.pbtxt
+
+# Install TensorRT and copy the Diffusion backend
+RUN pip install --no-cache-dir --extra-index-url https://pypi.nvidia.com tensorrt==10.0.1
+RUN git clone https://github.com/NVIDIA/TensorRT.git -b v10.0.1 --single-branch /tmp/TensorRT
+RUN cp -rf /tmp/TensorRT/demo/Diffusion /opt/tritonserver/backends/diffusion/
+#### End of Stable Diffusion modifications ####
+
+
+#### Start of Anyscale modifications ####
+RUN apt-get clean && apt-get update -y
+RUN apt-get install -y sudo tzdata supervisor openssh-client openssh-server rsync zip unzip git nfs-common
+
+# Delete the existing uid 1000 user so ray user can be created
+RUN existing_user=$(id 1000 | sed "s/uid=1000(//;s/) gid=1000.*//") && userdel "$existing_user"
+# ray user needs to be at uid 1000 and gid 100
+RUN useradd -ms /bin/bash -d /home/ray ray --uid 1000 --gid 100
+
+RUN sudo usermod -aG root ray
+RUN echo 'ray ALL=NOPASSWD: ALL' >> /etc/sudoers
```

Comment:
this stuff is super hacky -- is it necessary for all BYOD images?

Reply:
Yes, this is required. Anyscale assumes these is a user "ray" at uid 1000 and has sudo permission for it to do all kind of things 😞